dfget/core/downloader/p2p_downloader/power_client.go

```diff
@@ -119,7 +119,7 @@ func (pc *PowerClient) downloadPiece() (content *pool.Buffer, e error) {
 	peerPort := pc.pieceTask.PeerPort
 
 	// check that the target download peer is available
-	if dstIP != "" && dstIP != pc.node {
+	if dstIP != "" && dstIP != pc.node && pc.pieceTask.Path != pc.cfg.URL {
```

Comment:
Maybe use `pc.cdnSource != apiTypes.CdnSourceSource` is better? WDYT?

Reply:
There are three scenarios.
- `pc.cdnSource == apiTypes.CdnSourceSource && dstIP != supernodeIp`, connect check is very necessary. 
- Only `pc.pieceTask.Path != pc.cfg.URL`, need download piece from other peer, and `dstIP  != supernodeIp`, connect check is very necessary. 
- Only `pc.pieceTask.Path == pc.cfg.URL`, need download piece from source，and `dstIP  == supernodeIp`, connect check is not necessary.

Reply:
There are three scenarios.
- `pc.cdnSource == apiTypes.CdnSourceSource && dstIP != supernodeIp`, connect check is very necessary. 
- Only `pc.pieceTask.Path != pc.cfg.URL`, need download piece from other peer, and `dstIP  != supernodeIp`, connect check is very necessary. 
- Only `pc.pieceTask.Path == pc.cfg.URL`, need download piece from source，and `dstIP  == supernodeIp`, connect check is not necessary.